src/common/components/RichTextEditor/index.tsx

```diff
@@ -1,39 +1,75 @@
 import { useTheme } from '@mui/material'
-import { Plate } from '@udecode/plate-core'
+import {
+  createPlateEditor,
+  deserializeHtml as plateDeserializeHtml,
+  Plate,
+  serializeHtml as plateSerializeHtml,
+  TNode,
+  usePlateActions,
+} from '@udecode/plate-core'
 import { EditableProps } from 'slate-react/dist/components/editable'
 
-import React from 'react'
+import React, { forwardRef, useCallback, useImperativeHandle } from 'react'
 
 import { HighlightHTML } from '../HighlightHTML'
 import { Toolbar } from './components/Toolbar'
 import { INITIAL_CONTENT } from './constants'
 import { plugins } from './plugins'
-import { RichTextEditorProps } from './types'
-
-export const RichTextEditor: React.FC<RichTextEditorProps> = ({ id, defaultValue = INITIAL_CONTENT, onChange }) => {
-  const theme = useTheme()
-
-  const editableProps: EditableProps = {
-    id,
-    autoCapitalize: 'off',
-    autoComplete: 'off',
-    autoCorrect: 'off',
-    spellCheck: false,
-    placeholder: 'คุณคิดว่าวิชานี้เป็นอย่างไรบ้าง?',
-    style: {
-      padding: theme.spacing(2),
-      marginBottom: theme.spacing(2),
-      border: `1px solid ${theme.palette.divider}`,
-      borderBottomLeftRadius: theme.shape.borderRadius,
-      borderBottomRightRadius: theme.shape.borderRadius,
-    },
-  }
+import { RichTextEditorProps, RichTextEditorRef } from './types'
+
+const editor = createPlateEditor({
+  plugins: plugins,
+})
+
+export const RichTextEditor = forwardRef<RichTextEditorRef | undefined, RichTextEditorProps>(
+  ({ id, defaultValue = INITIAL_CONTENT, onChange }, ref) => {
+    const theme = useTheme()
+    const { resetEditor, setValue: setEditorValue } = usePlateActions(id)
+
+    const setValue = useCallback(
+      (nodes: TNode[]) => {
+        setEditorValue(nodes, id)
+        resetEditor(id)
```

Comment:
yo, what does `resetEditor(id)` do?
@saenyakorn i'm just curious

Reply:
Just `setEditorValue` can't display the change on the web (Dont know why). You need to reset the editor.

Reply:
ok, magic